src/main/java/org/openstreetmap/atlas/geography/atlas/items/complex/buildings/ComplexBuilding.java

```diff
@@ -58,6 +59,34 @@ protected ComplexBuilding(final AtlasEntity source)
         }
     }
 
+    /**
+     * @return The building's base height
+     */
+    public Optional<Distance> baseHeight()
+    {
+        Map<String, String> tags = getSource().getTags();
+        String minHeightTag = tags.get(MinHeightTag.KEY);
+        try
+        {
+            if (minHeightTag != null)
+            {
+                return Optional.of(HEIGHT_CONVERTER.convert(minHeightTag));
```

Comment:
Would it be possible to create a "getMinimumHeight" function that does:
```
try
{
    String minHeight = tags.get(MinHeightTag.KEY);

    if (minHeightTag != null)
    {
        return Optional.of(HEIGHT_CONVERTER.convert(minHeightTag));
    }
}
catch(...)
{
...
}
```

Then we could make 2 separate calls here and if the first call throw an exception, the second one can still be checked!

Reply:
Will cleanup this code, looks like its more relevant for height, not for min_height.